### ISSUE TITLE
Laravel 5.0 and 5.1 compatibility issue with lists()

### ISSUE THREAD

**Issue by wowzzangga** [kodeine/laravel-acl#60]
I'm currently working with Laravel v.5.0, and I have an error in the HasRole.php since this https://github.com/kodeine/laravel-acl/commit/8e5bcd4e0befdf9de6e5de498a613510e4873168 update. 

```FatalErrorException in HasRole.php line 35:
Call to a member function all() on array```

The lists() method still returns an array in v.5.0 whilst it no longer returns an array in v.5.1.

Could you make it compatible?

**Comment by kodeine** [comment]
@wowzzangga 
so instead of using `toArray()` if we cast it as (array) would it solve the issue?

**Comment by avlastenok** [comment]
as per laravel docs you would need 
to change
$this->roles->lists('slug')->toArray();
Into:
$this->roles ->lists('slug')->all();

**Comment by kodeine** [comment]
@avlastenok 
yeah but that `all()` method is not available for 5.0, for backward compatibility, we need to type cast it as array. Or you have anything else in mind?

**Comment by avlastenok** [comment]
But laravel 5.0 does have the all method
http://laravel.com/docs/5.0/eloquent#basic-usage Scroll down to retrieving all records

**Comment by kodeine** [comment]
@avlastenok 
`$permission->lists('name')->all()` wont work on 5.0, just checked.

**Comment by avlastenok** [comment]
Must be a error in the documents then, well I did not try it out on 5.0 just read the docs for it.

**Comment by kodeine** [comment]
@xkairu @avlastenok @wowzzangga 
I've pushed commits to branch 1.0, please try and let me know.

**Comment by xKairu** [comment]
https://github.com/kodeine/laravel-acl/blob/1.0/src/Kodeine/Acl/Traits/HasRole.php#L35
If the user has roles, return an empty array? Changing that line to this now returns properly: 
`return is_null($this->roles)`

https://github.com/kodeine/laravel-acl/blob/1.0/src/Kodeine/Acl/Helper/Helper.php#L24
For some reason, changing this line to `->all()` still returns a Collection, when it should fix all the issues?

https://github.com/kodeine/laravel-acl/blob/1.0/src/Kodeine/Acl/Traits/HasRole.php#L37
This doesn't return an array on 5.1. Changing the line to this seems to work, which I know will not work on 5.0. I'm just using this as a workaround for now:
`: $this->collectionAsArray($slugs)->all();`

**Comment by kodeine** [comment]
@xKairu 
I have fixed, it works on 5.0. I was missing the use trait in helpers file but now its added so collectionAsArray should work.

**Comment by xKairu** [comment]
Everything seems to be good now, thanks :+1:
